pkg/api/v1/resource/helpers.go

```diff
@@ -96,6 +97,47 @@ func GetResourceRequest(pod *v1.Pod, resource v1.ResourceName) int64 {
 	return totalResources
 }
 
+// PodResourceAllocations returns a dictionary of resources allocated to the containers of pod.
+func PodResourceAllocations(pod *v1.Pod) (allocations v1.ResourceList) {
+	allocations = v1.ResourceList{}
+	for _, container := range pod.Spec.Containers {
+		addResourceList(allocations, container.ResourcesAllocated)
+	}
+	// init containers define the minimum of any resource
+	for _, container := range pod.Spec.InitContainers {
+		maxResourceList(allocations, container.Resources.Requests)
+	}
+	return
+}
+
+// GetResourceAllocation finds and returns resource allocation for a specific resource.
```

Comment:
if simply the resource value is returned, the caller will need to adjust with the unit ( Mi, Gi, M, G etc ) which is error prone. return the rQuantity probably a better choice here.

Reply:
the use case for this is to get the actual value. if vertical scaling feature is on, then we would use this instead of [GetResourceRequest](https://github.com/futurewei-cloud/arktos/pull/343/files#diff-09e7de804ef61948e84b89d9dea88e51R73)